### DISCUSSION TITLE
Hugo causes HTML id collision for `search`

### DISCUSSION THREAD

**Discussion by gene1wood** [authelia/authelia#6071] [General]
### Version

v4.37.5

### Deployment Method

Docker

### Reverse Proxy

Caddy

### Description

(This isn't so much a "mistake in the documentation" which should go in the discussion area as it is an issue with authelia's use of Hugo)

The documentation generator interprets this line

https://github.com/authelia/authelia/blame/b18b6b6616d7d71e5125355c18c6e0a28c3dfd71/docs/content/en/configuration/first-factor/file.md#L75

which defines a heading called `search` and creates an `h3` tag with and `id="search"` so that links in the page to the anchor `#search` take you to that location in the document.

The problem is that Hugo also adds a search field at the top of every doc page that looks like this

```
<input id="search" class="form-control is-search" type="search" placeholder="Search..." aria-label="Search..." autocomplete="off">
```

As a result there's a collision with the `search` ID in the DOM and clicking links to the `#search` anchor tag don't take you to the search section of the document but instead to the search input field.

I think the easiest fix for this is to have Hugo generate a custom anchor for this heading

https://gohugo.io/content-management/cross-references/

to avoid the collision

### Reproduction

1. Visit this page https://www.authelia.com/configuration/first-factor/file/
2. Click the `search` header in the right hand table of contents
3. The page doesn't move

### Expectations

3. The page would skip to the `search` section of the document

### Build Information

```shell
n/a
```

### Logs (Authelia)

```shell
n/a
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have either included the complete debug / trace logs or the output of the build-info command if the logs are not relevant

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
Only issues related to the authelia application should be reported as issues.

**Comment by gene1wood** [comment]
Thanks for the fix to this in #6072
